### PR TITLE
Packer init: remove host and namespace guessing

### PR DESCRIPTION
**Pull request by azr**
```shellsession
❯ packerdev validate .
Error: Invalid plugin source string

  on formatted.pkr.hcl line 21, in packer:
  21:     comment-that-works = {
  22:       source  = "sylviamoss/comment"
  23:       version = "v0.2.19"
  24:     }

The "source" attribute must be in the format "hostname/namespace/name"

Error: Invalid plugin requirement

  on formatted.pkr.hcl line 20, in packer:
  20:     comment = "v0.1.3"

'comment = "v0.1.3"' plugin requirement calls are not possible. You must define
a whole block. For example:
comment = {
  source  = "github.com/hashicorp/comment"
  version = "v0.1.3"
}


exit status 1
```

### CODE REVIEW

File: website/content/docs/plugins/index.mdx [hunk 1]
```diff
@@ -151,37 +151,32 @@ to download it.
...
 
 The source address with all three components given explicitly is called the
 plugin's _fully-qualified address_. You will see fully-qualified address in
 various outputs, like error messages, but in most cases a simplified display
 version is used. Therefore you may see the shortened version `"myawesomecloud"`
-instead of `"github.com/hashicorp/myawesomecloud"`.
-
--> **Note:** We recommend using explicit source addresses for all plugins.
+instead of `"github.com/hashicorp/myawesomecloud"`. This will happen only when
+the prefix is `github.com/hashicorp`.
```
Comment (sylviamoss) [inline website/content/docs/plugins/index.mdx:179]: This is kind of confusing. Can we always make the display like `github.com/hashicorp/myawesomecloud`?

File: hcl2template/addrs/plugin.go [hunk 2]
```diff
@@ -34,20 +30,23 @@ func (p Plugin) String() string {
 // ForDisplay returns a user-friendly FQN string, simplified for readability. If
 // the plugin is using the default hostname, the hostname is omitted.
 func (p *Plugin) ForDisplay() string {
+	const (
```
Comment (sylviamoss) [inline hcl2template/addrs/plugin.go:33]: I would always display `hostname/namespace/type` here 🤔

**Review by sylviamoss — APPROVED** [review]
This looks good! 

Could you update the guide as well? Here https://packer-l6v09johj.vercel.app/guides/1.7-template-upgrade#source

File: website/content/docs/commands/init.mdx [hunk 3]
```diff
@@ -44,11 +44,10 @@ block :
 ```hcl
 packer {
   required_plugins {
-    myawesomecloud = {
+    happycloud = {
       version = ">= 2.7.0"
-      source = "azr/myawesomecloud"
+      source = "azr/happycloud"
```
[the remark this answers is not in the document]
Comment (azr) [reply website/content/docs/commands/init.mdx]: nice catch !

**Comment by azr** [comment]
Hey Moss, thanks for reviewing; I fixed the guide !

About `ForDisplay`, yeah you're right it's totally confusing now, I wanted to keep it to have consistency with Terraform; so that users would call the `amazon` and the `google` plugins instead of the `github.com/hashicorp/google` ones.

I'm going to remove that function then, the `String` one does just that.,

File: website/content/guides/1.7-template-upgrade.mdx [hunk 4]
```diff
@@ -135,10 +135,12 @@ in the rest of the template.
 
 Here it is a brief explanation of each field:
 - `version` - Should follow the [version constraints](/docs/templates/hcl_templates/blocks/packer#version-constraints).
-- `source` - Should have the plugin's organizational namespace on GitHub and its short name. Packer will be responsible
-for determining the full GitHub address.
-  For example, if the source is `sylviamoss/comment`, Packer will download the binaries from `github.com/sylviamoss/packer-plugin-comment`.
-  To learn more about the source field, check out the [Source Address](/docs/plugins#source-addresses) documentation.
+- `source` - Should have the plugin's organizational namespace on GitHub and its
+  short name. Packer will be responsible for determining the full GitHub
```
Comment (sylviamoss) [inline website/content/guides/1.7-template-upgrade.mdx:139]:
```suggestion
- `source` - Should have the GitHub hostname, the plugin's organizational namespace, and its short name. 
Packer will be responsible for determining the full GitHub
```
Do you think we should add the hostname here?

**Review by sylviamoss — APPROVED** [review]
Thanks for updating the guide!

File: website/content/guides/1.7-template-upgrade.mdx (hunk 4 above)
Comment (azr) [reply website/content/guides/1.7-template-upgrade.mdx:139]: yes !

**Review by SwampDragons — APPROVED** [review]

**Pull request merged**